src/main/java/com/sivalabs/springblog/post/web/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,40 @@
+package com.sivalabs.springblog.post.web.exception;
+
+import com.sivalabs.springblog.post.domain.PostNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.net.URI;
+import java.time.Instant;
+
+@RestControllerAdvice
```

Comment:
As we are not building REST API, this need to be @ControllerAdvice and from the @ExceptionHandler methods we need to render error page views instead of returning JSON responses.